crates/oxc_linter/src/rules/eslint/sort_keys.rs

```diff
@@ -202,7 +202,7 @@ fn alphanumeric_sort(arr: &mut [String]) {
 }
 
 fn natural_sort(arr: &mut [String]) {
-    arr.sort_by(|a, b| {
+    arr.sort_unstable_by(|a, b| {
```

Comment:
`Ordering::Equal` is returned only when it's a same value, so we can use unstable sort here.